Updated surface tracker documentation. (#228)
pupil-labs/pupil-docs: user-docs/pupil-capture.md

Markers can be printed on paper, stickers, or displayed on a screen.

<aside class="notice">
  <strong>Note</strong> - When displaying or printing markers, ensure that a white border remains around the marker! 
  The border should be at least 1.2 times as wide as the grid size of the marker. 
  One way to achieve this is to present the marker on a white (or very light colored) background with sufficient padding.
</aside>

The design of our markers was greatly inspired by the [ArUco marker tracking library](http://www.uco.es/investiga/grupos/ava/node/26).
However our markers use 5x5 grid instead of the 7x7 grid ArUco uses. 
This allows us to make smaller markers that can still be detected well.
The 5x5 design allows for a total of 63 unique markers.  

#### Preparing your Environment

A surface can be based on one or more markers.
The markers need to be placed in close proximity or within your desired AOI.
If your AOI is for example a computer monitor, you could display your markers in the corners of the screen or place them somewhere on the bezel.

> If your AOI is a magazine page you could place the markers in the corners of the page or anywhere else on the page where they are not occluding.

Reviewer: ```suggestion
If your AOI is a magazine page, you could place the markers in the corners of the page, or anywhere else on the page where they are not occluding the content.
```

> When placing your markers please follow the following rules:

Reviewer: ```suggestion
When placing your markers please follow the guidelines:
```

> *   All markers of a surface need to lie within the same plane.

Reviewer: ```suggestion
*   All markers of a surface need to lie within the same two dimensional plane.
```

*   An individual marker can be part of multiple surfaces.
*   The used markers need to be unique, i.e. you may not use multiple instances of the same marker in your environment.
*   Using more markers to define a surface yields greater robustness in the tracking of that surface.
*   Surfaces defined with more than 2 markers are detected even if some markers lie outside of the camera image or are obscured.

#### Defining a Surface 

> {{< video-webm src="/videos/surface-tracking/srf-tracking.webm" >}}

> Surfaces can be defined with Pupil Capture in real-time, or offline with Pupil Player. 

Reviewer: ```suggestion
Surfaces can be defined with Pupil Capture in real-time, or post-hoc with Pupil Player. 
```

In both cases the necessary steps are as follows:

*   Prepare your environment as described above.
*   Turn on the `Surface Tracker` plugin .
*   Make sure the camera is pointing at your AOI and the markers are well detected.

>     In the offline case seek to a frame that contains a good view of your desired AOI.

Reviewer: ```suggestion
    In the post-hoc case (using Pupil Player) seek to a frame that contains a good view of your desired AOI.
```

*   Add a new surface by clicking the `Add surface` button.
*   Give your surface a name.
*   Click the `edit surface` button and move the corners of your surface into the desired position.

>     In the online case this is much easier if you freeze the video by clicking the `Freeze Scene` button.

Reviewer: ```suggestion
    In the real-time case (using Pupil Capture) this is much easier if you freeze the video by clicking the `Freeze Scene` button.
```

*   If markers have been erroneously added or left out, click the `add/remove markers` button and afterwards onto the according marker to add/remove them from your surface.

#### Reusing Surface Definitions

Your surfaces are automatically saved in a file called `surface_definitions` in the `pupil_capture_settings` directory. 
If you restart Pupil Capture or the Surface Tracker plugin, your surface definitions from previous sessions will be loaded.

> Further the `surface_definitions` file is copied into each recording folder as well, so you have access to your surface definitions in Pupil Player. 

Reviewer: ```suggestion
The `surface_definitions` file is copied into each recording folder as well, so you will have access to your surface definitions in Pupil Player. 
```

> You can copy&paste this file to move definitions from one session or recording to another.

Reviewer: ```suggestion
You can copy & paste this file to move definitions from one session or recording to another.
```

#### Gaze Heatmaps for Surfaces

> It is possible to display gaze heatmaps for each surface by enabling `Show Heatmap` in the `Surface Tracker` menu.

Reviewer: ```suggestion
You can display gaze heatmaps for each surface by enabling `Show Heatmap` in the `Surface Tracker` menu.
```

Two heatmap modes are supported:
*   `Gaze within each surface`: Visualizes the distribution of gaze points that lie within each surface. 
*   `Gaze across different surfaces`: Color codes the surfaces to visualize the amount of time spend gazing on each surface in relation to other surfaces.

> Red color stands for a lot of gaze points or time spend, while blue color stands for few gaze points or little time spend. 

Reviewer: ```suggestion
Red color represents a lot of gaze points or time spent. Blue color represents few gaze points or little time spent. 
```
